alerting/src/main/kotlin/org/opensearch/alerting/transport/TransportAcknowledgeChainedAlertAction.kt

```diff
@@ -0,0 +1,322 @@
+import org.opensearch.commons.alerting.action.AlertingActions
+import org.opensearch.commons.alerting.action.GetWorkflowRequest
+import org.opensearch.commons.alerting.action.GetWorkflowResponse
+import org.opensearch.commons.alerting.model.Alert
+import org.opensearch.commons.alerting.model.CompositeInput
+import org.opensearch.commons.alerting.model.DataSources
+import org.opensearch.commons.alerting.model.Workflow
+import org.opensearch.commons.alerting.util.optionalTimeField
+import org.opensearch.commons.utils.recreateObject
+import org.opensearch.core.xcontent.NamedXContentRegistry
+import org.opensearch.core.xcontent.XContentParser
+import org.opensearch.index.query.QueryBuilders
+import org.opensearch.rest.RestRequest
+import org.opensearch.rest.RestStatus
+import org.opensearch.search.builder.SearchSourceBuilder
+import org.opensearch.search.fetch.subphase.FetchSourceContext
+import org.opensearch.tasks.Task
+import org.opensearch.transport.TransportService
+import java.time.Instant
+import java.util.Locale
+
+private val log = LogManager.getLogger(TransportAcknowledgeChainedAlertAction::class.java)
+private val scope: CoroutineScope = CoroutineScope(Dispatchers.IO)
+
+class TransportAcknowledgeChainedAlertAction @Inject constructor(
+    transportService: TransportService,
+    val client: Client,
+    clusterService: ClusterService,
+    actionFilters: ActionFilters,
+    val settings: Settings,
+    val xContentRegistry: NamedXContentRegistry,
+    val transportGetMonitorAction: TransportGetMonitorAction,
+    val transportGetWorkflowAction: TransportGetWorkflowAction,
+) : HandledTransportAction<ActionRequest, AcknowledgeAlertResponse>(
+    AlertingActions.ACKNOWLEDGE_CHAINED_ALERTS_ACTION_NAME,
+    transportService,
+    actionFilters,
+    ::AcknowledgeChainedAlertRequest
+) {
+    @Volatile
+    private var isAlertHistoryEnabled = AlertingSettings.ALERT_HISTORY_ENABLED.get(settings)
+
+    init {
+        clusterService.clusterSettings.addSettingsUpdateConsumer(AlertingSettings.ALERT_HISTORY_ENABLED) { isAlertHistoryEnabled = it }
+    }
+
+    override fun doExecute(
+        task: Task,
+        AcknowledgeChainedAlertRequest: ActionRequest,
+        actionListener: ActionListener<AcknowledgeAlertResponse>,
+    ) {
+        val request = AcknowledgeChainedAlertRequest as? AcknowledgeChainedAlertRequest
+            ?: recreateObject(AcknowledgeChainedAlertRequest) { AcknowledgeChainedAlertRequest(it) }
+        client.threadPool().threadContext.stashContext().use {
+            scope.launch {
+                try {
+                    val getWorkflowResponse: GetWorkflowResponse =
+                        transportGetWorkflowAction.client.suspendUntil {
+                            val getWorkflowRequest = GetWorkflowRequest(workflowId = request.workflowId, method = RestRequest.Method.GET)
+
+                            execute(AlertingActions.GET_WORKFLOW_ACTION_TYPE, getWorkflowRequest, it)
+                        }
```

Comment:
This will have issues with RBAC. You can refer to this to replace this logic with some minor modifications.

https://github.com/opensearch-project/alerting/blob/main/alerting/src/main/kotlin/org/opensearch/alerting/transport/TransportIndexMonitorAction.kt#L574-L593

Reply:
replaced with getRequest from index